### PR TITLE
Pearson and Spearman Correlation are mainly used for regression problem.

### PR DESCRIPTION
**Pull request by richarddwang**
Currently the two metric is `dim_argmax=-1`, but they are mainly used for regression problem (e.g. GLUE's stsb task), which pred is not logits but scores with the same shape with targets.

I move the two to Regression section, and make default `dim_argmax=None` for them, but users can still specify `dim_argmax=-1` if they want to use them in classification problem.

### CODE REVIEW

**Comment by jph00** [comment]
Seems you removed `scs_to_fastai`? Could you please add tests for these when you move them?

**Comment by richarddwang** [comment]
Tests for Pearson and Spearman correlation coefficient has been added.

**Comment by jph00** [comment]
Thanks! I think `scs_to_fastai` is still removed though?

**Comment by richarddwang** [comment]
Seems that I misunderstand something.

I add `scs_to_fastai` with `PearsonCorrCoef` and `SpearmanCorrCoef` metrics in a pr. And now I found `scs_to_fastai` is actually not useful and redundant so I remove it.

Is there any things I need to do now ?

**Comment by jph00** [comment]
Yes please put it back. fastai has a public API. Only things with a `_` suffix are private API. We don't remove things from the public API if it's not used internally - users are still using it externally.

**Comment by jph00** [comment]
...well, I guess it's simple enough that maybe it's not a problem. I'll merge the PR, and if folks complain that they were using it, we can always put it back :)

**Comment by richarddwang** [comment]
Oh, I see. 
And thank you !
